docs/tasks/istio-installation.md

```diff
@@ -7,31 +7,106 @@ layout: docs
+This page shows how to install and configure Istio in a Kubernetes cluster.
 {% endcapture %}
 
 {% capture prerequisites %}
-* Do this.
-* Do this too.
+* The following instructions assume you have access to a Kubernetes cluster. To install Kubernetes locally, try [minikube](https://kubernetes.io/docs/getting-started-guides/minikube/).
+* If you are using [Google Container Engine](https://cloud.google.com/container-engine), please make sure you are using static client certificates before fetching cluster credentials:
+```bash
+gcloud config set container/use_client_certificate True
+gcloud container clusters get-credentials cluster-2 --zone us-central1-a --project istio-demo
+```
+* Ensure the curl command is present, or
 {% endcapture %}
 
 {% capture steps %}
-## Doing ...
+## Installing on an existing cluster
+For the Alpha release, Istio must be installed in the same Kubernetes namespace as the applications. Instructions below will deploy Istio in the "default" namespace. They can be modified for deployment in a separate namespace.
 
-1. Do this.
-1. Do this next. Possibly read this [related explanation](...).
+1. a. Download and extract the [istio installation files](https://raw.githubusercontent.com/istio/istio/master/releases/istio-alpha.tar.gz), or\
+b. Clone the github [istio](https://github.com/istio/istio) repository:
+```bash
+git clone https://github.com/istio/istio
+```
+1. Change directory to istio:
+```bash
+cd istio
+```
+2. Install the Istio core components (Istio-Manager, Mixer, and Ingress-Controller):
+```bash
+kubectl apply -f ./kubernetes/istio.yaml
+```
+1. Source the Istio configuration file:
+```bash
+source istio.VERSION
+```
+3. Download one of the [`istioctl`]({{site.bareurl}}/docs/reference/istioctl.html) client binaries corresponding to your OS: `istioctl-osx`, `istioctl-win.exe`,
+`istioctl-linux`, targeted at Mac, Windows or Linux users respectively. For example, run the following commands on a Mac system:
+```bash
+curl ${ISTIOCTL_URL}/istioctl-osx > /usr/local/bin/istioctl
+chmod +x /usr/local/bin/istioctl
+```
+Istioctl is needed to inject Envoy as a sidecar proxy. It also provides a convenient CLI for creating routing rules and policies.
+4. Deploy your application:
+```bash
+kubectl create -f <(istioctl kube-inject -f <your-app-spec>.yaml)
+```
+The [kube-inject]({{site.bareurl}}/docs/reference/istioctl.html##kube-inject) tool will automatically inject an Envoy container in the pod running the application.
+Alternatively, deploy one of the samples applications, for instance [bookinfo]({{site.bareurl}}/docs/samples/bookinfo.html).
+
+5. Optionally: to view metrics collected by Mixer, install [Prometheus](https://prometheus.io), [Grafana](http://staging.grafana.org) and ServiceGraph addons:
+```bash
+kubectl apply -f ./kubernetes/addons/grafana.yaml
+kubectl apply -f ./kubernetes/addons/prometheus.yaml
+kubectl apply -f ./kubernetes/addons/servicegraph.yaml
+```
```

Comment:
I don't understand the earlier comment about service graph. Why do we still need the service graph in the addons? It does not work (the last I tried it). Besides, When we have zipkin in place, it will already give us the service graph.